NRLMMD-GEOIPS/geoips
# 35 create glm reader

# Reviewer Checklist

* [ ] Required ***unit tests*** added and pass for new/modified functionality

# Related Issues
fixes #35 

# Testing Instructions
You can run tests/integration_tests/full_test.sh to ensure that everything is still working correctly. To see output for the 'test_glm.sh' script, simply run ./tests/scripts/test_glm.sh and click the output image to see what has been produced.

# Summary
Kyle Hilburn works with GLM data, and has requested that we build a reader for that data
so he can implement his routines in GeoIPS. This branch has not only created that
reader, but also includes products, a feature_annotator, output_formatter, and a basic
algorithm to visualize GLM area products. We also modified mpl_utils.py to include a
patch adder, which adds patches to the appropriate figure in the correct Coordinate
Reference System (CRS), with the appropriate colormap applied.

    modified: geoips/geoips/image_utils/mpl_utils.py
    modified: geoips/geoips/plugins/modules/procflows/single_source.py
    modified: geoips/pyproject.toml
    added: geoips/geoips/plugins/modules/algorithms/glm_alg.py
    added: geoips/geoips/plugins/modules/output_formatters/shape_patches.py
    added: geoips/geoips/plugins/modules/readers/glm_netcdf.py
    added: geoips/geoips/plugins/yaml/products/glm.yaml
    added: geoips/tests/scripts/test_glm.sh

# Output
![20230101 000000 GOES-18 glm flash_area goes_east 100p00 gov nesdis noaa 10p0](https://github.com/NRLMMD-GEOIPS/geoips/assets/62626811/9a44005c-01d1-4eb7-9359-d6880483d298)
![20230101 000000 GOES-18 glm group_area goes_east 100p00 gov nesdis noaa 10p0](https://github.com/NRLMMD-GEOIPS/geoips/assets/62626811/8ecbc7a0-b369-49a5-a4f2-e65072534ba2)
